### PR TITLE
Release allocated memory by mbed TLS on json_sign

### LINKED ISSUE #54500: jwt: memory allocation problem after multiple jwt_sign calls
**Describe the bug**

When called multiple times `jwt_sign()`, i got memory allocation problems with mbedTLS.
I'm using a NRF52840 module

**To Reproduce**

Create a simple project that frequently use **jwt_sign()**

```
#define EXP_TIMESTAMP 1675698491
#define IAT_TIMESTAMP 1675693491
#define AUD_JWT "myAut"

static uint8_t token[512];
struct jwt_builder jb;

const unsigned char private_der[] = {}; //Insert a DER private key
unsigned int private_der_len = 1216;

int main()
{
     jwt_init_builder(&jb, token, sizeof(token));
     jwt_add_payload(&jb, EXP_TIMESTAMP, IAT_TIMESTAMP,AUD_JWT);

    while(true)
   {
      int ret = jwt_sign(&jb, private_der, private_der_len);
      if(ret < 0)
      {
         LOG_ERR("%d",ret);
      }
      k_msleep(1000);
   }
}
```

**Expected behavior**
After some executions, the **jwt_sign** will throw memory allocation error and will keep it until the application restart.

**Environment:**
 - OS: Windows
 - Toolchain: Zephyr 3.2.0

### PR DESCRIPTION
**Pull request by Yago-Caetano**
Fixes: #54500 

This commit release allocated memory by mbedTLS on `json_sign` function

Signed-off-by: Yago Caetano <yago_caetano@outlook.com>

### CODE REVIEW

**Comment by MaureenHelm** [comment]
please rebase

File: subsys/jwt/jwt.c [hunk 1]
```diff
@@ -229,13 +230,15 @@ int jwt_sign(struct jwt_builder *builder,
 			      sig, sig_len, &sig_len,
```
Comment (joerchan) [inline subsys/jwt/jwt.c:230]: Speaking of error handling, the call to `mbedtls_sha256` does not check the return value.
@Yago-Caetano Could you check that?

**Review by joerchan — CHANGES_REQUESTED** [review]
The changes looks good, but the merge-commit needs to be removed.

File: subsys/jwt/jwt.c [hunk 2]
```diff
@@ -221,22 +222,30 @@ int jwt_sign(struct jwt_builder *builder,
 	 * The '0' indicates to mbedtls to do a SHA256, instead of
 	 * 224.
 	 */
-	mbedtls_sha256(builder->base, builder->buf - builder->base,
+	res = mbedtls_sha256(builder->base, builder->buf - builder->base,
 		       hash, 0);
+		       
+	if(res != 0){
```
Comment (joerchan) [inline subsys/jwt/jwt.c:228]:
```suggestion
	if (res != 0) {
```

**Comment by joerchan** [comment]
@Yago-Caetano The zephyr workflow does not allow merge-commits, you now have two in your history. You need to use rebase instead.

**Pull request closed without merging**